superset/views/core.py

```diff
@@ -244,14 +244,32 @@ def _deserialize_results_payload(
             return json.loads(payload)  # type: ignore
 
 
+SQLTable = Table(
+    "tables",
+    Model.metadata,  # pylint: disable=no-member
+    Column("id", Integer, primary_key=True),
+    Column("database_id", Integer, ForeignKey("dbs.id")),
+    extend_existing=True,
+)
+
+
 class SliceFilter(BaseFilter):
     def apply(self, query, func):  # noqa
         if security_manager.all_datasource_access():
             return query
-        perms = security_manager.user_view_menu_names("datasource_access")
+        database_perms = security_manager.user_view_menu_names("database_access")
```

Comment:
what do you think about materializing db permissions in the tables and slices?
this would simplify the checks significantly.